src/utils/card.ts

```diff
@@ -0,0 +1,187 @@
+// if not, forces user to pass phone number again to create new JWT pair in sessionStorage.
+export async function defineUserStatus(): Promise<Status> {
+  const sessionRefreshToken = sessionStorage.getItem('refresh-token');
+  let sessionAccessToken = sessionStorage.getItem('access-token');
+
+  if (!(sessionAccessToken && sessionRefreshToken)) {
+    return emptyStatusFields();
+  }
+
+  if (isAccessTokenExpired(sessionAccessToken)) {
+    const accessToken = await getUpdatedJwtPair(sessionRefreshToken);
+
+    if (accessToken) {
+      sessionAccessToken = accessToken;
+    } else {
+      return emptyStatusFields();
+    }
+  }
+
+  const { kycStatus, verificationStatus } = await getUserStatus(sessionAccessToken);
+
+  return { kycStatus, verificationStatus };
+}
+
+async function getUpdatedJwtPair(refreshToken: string | null): Promise<string | null> {
+  const soraNetwork = store.state.wallet.settings.soraNetwork || WALLET_CONSTS.SoraNetwork.Test;
+  const { apiKey } = soraCard(soraNetwork).authService;
+  const buffer = Buffer.from(apiKey);
+
+  try {
+    const response = await fetch('https://api-auth-test.soracard.com/RequestNewAccessToken', {
+      method: 'POST',
+      headers: {
+        Authorization: `Basic ${buffer.toString('base64')}, Bearer ${refreshToken}`,
+      },
+    });
+
+    if (response.status === 200 && response.ok === true) {
+      const accessToken = response.headers.get('accesstoken');
+      const expirationTime = response.headers.get('expirationtime');
+
+      if (accessToken && expirationTime) {
+        sessionStorage.setItem('access-token', accessToken);
+        sessionStorage.setItem('expiration-time', expirationTime);
+      }
+
+      return accessToken;
+    }
+  } catch (error) {
+    console.error('[SoraCard]: Error while getting new JWT pair', error);
+  }
+
+  return null;
+}
+
+async function getUserStatus(accessToken: string | null): Promise<Status> {
+  if (!accessToken) return emptyStatusFields();
+
+  try {
+    const result = await fetch('https://sora-card.sc1.dev.sora2.soramitsu.co.jp/kyc-last-status', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    });
+
+    const lastRecord = await result.json();
+    const verificationStatus: VerificationStatus = lastRecord.verification_status;
+    const kycStatus: KycStatus = lastRecord.kyc_status;
+
+    if (Object.keys(VerificationStatus).includes(verificationStatus) && Object.keys(KycStatus).includes(kycStatus)) {
+      return { verificationStatus, kycStatus };
+    }
+
+    return emptyStatusFields();
+  } catch (error) {
+    console.error('[SoraCard]: Error while getting KYC and verification statuses', error);
+    return emptyStatusFields();
+  }
+}
+
+const isAccessTokenExpired = (accessToken: string): boolean => {
+  try {
+    const decoded: JwtPayload = jwtDecode(accessToken);
+
+    if (decoded.exp) {
+      if (Date.now() <= decoded.exp * 1000) {
```

Comment:
is it number? decoded.exp

Reply:
I mean are we sure that it's number?

Reply:
yes
<img width="571" alt="Screenshot 2023-01-23 at 19 01 13" src="https://user-images.githubusercontent.com/27702061/214087321-e6f1358f-5e80-4107-a0d3-d2a99d03a6a2.png">